### PR TITLE
Move to version catalogs and properties-based configuration

### PR DESCRIPTION
**Pull request by zml2008**
With these changes, no regular version bumps should require modifying the project's buildscripts directly. This should allow buildscript compilation to be cached with less invalidation.

### CODE REVIEW

File: bom/build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 plugins {
-  id("java-platform")
+  `java-platform`
```
Comment (kashike) [inline bom/build.gradle.kts]: Not a fan of this change.

File: build-logic/settings.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      mavenContent { snapshotsOnly() }
+    }
+  }
+  
+  versionCatalogs {
+    register("libs") {
+      from(files("../gradle/libs.versions.toml")) // include from parent project
+    }
+  }
+}
```
Comment (kashike) [inline build-logic/settings.gradle.kts]: newline

**Review by kezz — APPROVED** [review]
I suppose plugins in the common conventions build logic can't use typesafe accessors still right?

**Comment by zml2008** [comment]
Unfortunately no -- since convention plugin plugin blocks can't actually affect the classpath, they're just for application.

**Pull request merged**